src/frontends/tensorflow/src/graph_iterator_saved_model.hpp

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <map>
+
+#include "graph_iterator_proto.hpp"
+#include "openvino/util/file_util.hpp"
+#include "saved_model.pb.h"
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+
+struct VIBlock;
+
+template <typename T>
+std::basic_string<T> getSMName() {}
+template <typename T>
+std::basic_string<T> getVIName() {}
+
+template <>
+std::basic_string<char> getSMName<char>();
+template <>
+std::basic_string<char> getVIName<char>();
+
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+template <>
+std::basic_string<wchar_t> getSMName<wchar_t>();
+template <>
+std::basic_string<wchar_t> getVIName<wchar_t>();
+#endif
+
+// Stores information about variables index
+class SMVariablesIndex {
+    int32_t totalShards;
+    std::map<std::string, std::vector<char>> varIndex;
+    std::map<int32_t, std::shared_ptr<std::ifstream>> dataFiles;
+    std::map<std::string, std::string> varMap;
+
+public:
+    /// \brief Reads variables from opened variable index file. Can cause an asserts in case of issues.
+    /// \param vi_stream Opened stream file, file pointer doesn't matter, it will be rewind internally.
+    /// \param path A path to file with variables data
+    /// \returns Returns true in case of everything loads successfully, false otherwise
+    bool readVariables(std::ifstream& vi_stream, const std::string& path);
```

Comment:
let us have method names aligned with other classes names: `read_variables`

Reply:
should this method be constant, please add `const`, if yes

Reply:
It is not code-style, it is a concept of C++ to mark methods like `const` if they are not going to change members. Please track other methods and classes.

Your PR is not ready because it misses tests, but I started reviewing PR in advance even:)

Reply:
I'm talking about naming not about const keyword.